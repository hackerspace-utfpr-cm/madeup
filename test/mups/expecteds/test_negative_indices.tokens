1(0-0) ID: a
1(2-2) ASSIGN: =
1(4-4) LEFT_CURLY_BRACE: {
1(5-5) INTEGER: 1
1(6-6) COMMA: ,
1(8-8) INTEGER: 2
1(9-9) COMMA: ,
1(11-11) INTEGER: 3
1(12-12) COMMA: ,
1(14-14) INTEGER: 4
1(15-15) RIGHT_CURLY_BRACE: }
1(16-16) NEWLINE: \n
2(17-21) ID: debug
2(23-23) ID: a
2(24-24) LEFT_BRACKET: [
2(25-26) INTEGER: -1
2(27-27) RIGHT_BRACKET: ]
2(29-30) EQUAL_TO: ==
2(32-32) INTEGER: 4
2(33-33) NEWLINE: \n
3(34-38) ID: debug
3(40-40) ID: a
3(41-41) LEFT_BRACKET: [
3(42-43) INTEGER: -2
3(44-44) RIGHT_BRACKET: ]
3(46-47) EQUAL_TO: ==
3(49-49) INTEGER: 3
3(50-50) NEWLINE: \n
4(51-55) ID: debug
4(57-57) ID: a
4(58-58) LEFT_BRACKET: [
4(59-60) INTEGER: -3
4(61-61) RIGHT_BRACKET: ]
4(63-64) EQUAL_TO: ==
4(66-66) INTEGER: 2
4(67-67) NEWLINE: \n
5(68-72) ID: debug
5(74-74) ID: a
5(75-75) LEFT_BRACKET: [
5(76-77) INTEGER: -4
5(78-78) RIGHT_BRACKET: ]
5(80-81) EQUAL_TO: ==
5(83-83) INTEGER: 1
5(84-84) NEWLINE: \n
6(85-85) END_OF_FILE: END OF FILE
