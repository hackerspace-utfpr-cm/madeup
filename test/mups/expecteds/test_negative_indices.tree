(block (define-variable a (ARRAYLITERAL (INTEGER 1) (INTEGER 2) (INTEGER 3) (INTEGER 4))) (call debug (== (SUBSCRIPT (call-with-names a) (INTEGER -1)) (INTEGER 4))) (call debug (== (SUBSCRIPT (call-with-names a) (INTEGER -2)) (INTEGER 3))) (call debug (== (SUBSCRIPT (call-with-names a) (INTEGER -3)) (INTEGER 2))) (call debug (== (SUBSCRIPT (call-with-names a) (INTEGER -4)) (INTEGER 1))))
