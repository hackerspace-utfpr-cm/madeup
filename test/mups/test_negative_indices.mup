a = {1, 2, 3, 4}
debug a[-1] == 4
debug a[-2] == 3
debug a[-3] == 2
debug a[-4] == 1
