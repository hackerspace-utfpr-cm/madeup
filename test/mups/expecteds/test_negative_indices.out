a[-1] == 4: true
a[-2] == 3: true
a[-3] == 2: true
a[-4] == 1: true
